### PR TITLE
Poc add zombienet

### PR DESCRIPTION
**Pull request by pepoviola**
Hi @wirednkod, I open this as `draft` to check how we can integrate the test you need in `zombienet`. At the moment I switch to use `native` provider in the `ci` since `podman` just hungs spawning the pods. 
If you have test that needs the `chain-spec` file and the `endpoints` of the nodes, we can add those to the `feature` file and will run inside the test-runner.

Thanks!

### CODE REVIEW

**Comment by tomaka** [comment]
For context, it's not the Polkadot nodes that we want to test. We want to test whether substrate-connect successfully synchronizes with the chain that the Polkadot nodes are running.

**Comment by wirednkod** [comment]
Thank you @pepoviola . I will create a simple test for implementing what @tomaka describes and use this PR for testing purposes if you are ok with that.

> If you have test that needs the chain-spec file and the endpoints of the nodes, we can add those to the feature file and will run inside the test-runner.

For that test the chainspecs (which should contain the spawned bootnodes) will be needed - I will check the repo of zombienet to figure out how to do this.

**Comment by pepoviola** [comment]
Thanks for the feedback @tomaka / @wirednkod, I use `polkadot` and the base `smoke test` from `zombienet` just to check how to integrate `zombienet` in GH Actions. @wirednkod ping me to work in your test case 👍 

Thanks!

File: .zombienet-tests/0001-custom.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+const polkadotApi = require("@polkadot/api");
+
```
Comment (wirednkod) [inline .zombienet-tests/0001-custom.js:2]: const { ScProvider } = require("@substrate/connect");

File: .zombienet-tests/0001-custom.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+const polkadotApi = require("@polkadot/api");
```
Comment (wirednkod) [inline .zombienet-tests/0001-custom.js:1]: replace that import with:
```javascript
const customChainSpec = require( "./config.json" )
const { ScProvider } = require("@substrate/connect");
```

File: .zombienet-tests/0001-custom.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+async function connect(apiUrl, types) {
+    const provider = new polkadotApi.WsProvider(apiUrl);
```
Comment (wirednkod) [inline .zombienet-tests/0001-custom.js:4]: replace the provider with:
```javascript
async function connect(customChainSpec, types) {
     const provider = ScProvider(customChainSpec)
```

File: .zombienet-tests/0001-custom.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    const api = new polkadotApi.ApiPromise({ provider, types });
+    await api.isReady;
+    return api;
+}
+
+async function run(nodeName, networkInfo, args) {
+    const {wsUri, userDefinedTypes} = networkInfo.nodesByName[nodeName];
+    const api = await connect(wsUri, userDefinedTypes);
```
Comment (wirednkod) [inline .zombienet-tests/0001-custom.js:12]: replace:
```javascript
    const api = await connect(wsUri, userDefinedTypes);
```
with:
```javascript
    const api = await connect(JSON.strigify(customChainSpec), userDefinedTypes);  
```

Comment (pepoviola) [reply .zombienet-tests/0001-custom.js:12]: `customChainSpec   should be an `string`?

File: .zombienet-tests/0001-custom.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  const api = new polkadotApi.ApiPromise({ provider, types })
+  await api.isReady
+  return api
+}
+
+async function run(nodeName, networkInfo) {
+  const { userDefinedTypes } = networkInfo.nodesByName[nodeName]
+  const customChainSpec = require(networkInfo.chainSpecPath)
+
+  // TODO: forkId generate an error in smoldot-light
```
Comment (tomaka) [inline .zombienet-tests/0001-custom.js:15]: Please tell me these kind of things

Comment (tomaka) [reply .zombienet-tests/0001-custom.js:15]: https://github.com/paritytech/smoldot/issues/1955

Comment (tomaka) [reply .zombienet-tests/0001-custom.js:15]: Smoldot v0.5.15 (published in ~30mn) fixes this.

Comment (pepoviola) [reply .zombienet-tests/0001-custom.js:15]: Thanks @tomaka! I will remove this from the test.

File: .zombienet-tests/0001-custom.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  return api
+}
+
+async function run(nodeName, networkInfo) {
+  const { userDefinedTypes } = networkInfo.nodesByName[nodeName]
+  const customChainSpec = require(networkInfo.chainSpecPath)
+
+  const api = await connect(JSON.stringify(customChainSpec), userDefinedTypes)
+  // add 30s sleep to give time to sync
+  await new Promise((resolve) => setTimeout(resolve, 30000))
```
Comment (tomaka) [inline .zombienet-tests/0001-custom.js:16]: This normally isn't needed

File: .zombienet-tests/0002-getHeaders.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+async function run(nodeName, networkInfo) {
+  const { userDefinedTypes } = networkInfo.nodesByName[nodeName]
+  const customChainSpec = require(networkInfo.chainSpecPath)
+  console.log("bootnodes", customChainSpec.bootNodes)
+
+  const api = await connect(JSON.stringify(customChainSpec), userDefinedTypes)
+  // add 20s sleep to give time to sync
+  await new Promise((resolve) => setTimeout(resolve, 20000))
```
Comment (tomaka) [inline .zombienet-tests/0002-getHeaders.js:17]: Same remark

Comment (wirednkod) [reply .zombienet-tests/0002-getHeaders.js:17]: We need a slight timeout for pausing the process until Chain is initialised.
If this timeout is removed then the following error shows:
```
2022-02-03 17:22:25        RPC-CORE: subscribeNewHeads(): Header:: Chain is not initialised
2022-02-03 17:22:25             DRR: Chain is not initialised
2022-02-03 17:22:25        RPC-CORE: subscribeNewHeads(): Header:: Chain is not initialised
unhandledRejection
Error: Chain is not initialised
```

Comment (pepoviola) [reply .zombienet-tests/0002-getHeaders.js:17]: We can use the `sleep x` command in the `feature` file instead of having in this scripts.

**Comment by wirednkod** [comment]
A new [PR](https://github.com/paritytech/substrate-connect/pull/706) is addressing this.

**Pull request closed without merging**